examples/reaction_roles.py

```diff
@@ -0,0 +1,34 @@
+"""Uses a messages to add and remove roles through reactions."""
+
+import discord
+from discord.ext import commands
+
+# This bot requires the members and reactions intensions.
+intents = discord.Intents.default()
+intents.members = True
+bot = commands.Bot(command_prefix=commands.when_mentioned_or("!"), intents=intents)
+
+role_message_id = 0  # ID of message that can be reacted to to add role
+emoji_to_role = {
+    "👍": 0,  # ID of role associated with thumbs up emoji
+    "test": 0  # ID of role associated with custom emoji 'test'
+}
```

Comment:
This should probably use the full emoji form not just the plain name, including the ID. Alternatively you can use `discord.PartialEmoji` as the key but this might seem too much for the passer-by.